webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/CreateNodeData.scala

```diff
@@ -237,6 +217,26 @@ object Creation {
     final case class OnAcceptation(message: String)                                          extends CreationError {
       override def errorMsg: String = s"[accept] ${message}"
     }
+
+    implicit val nodeValidationErrorEncoder: JsonEncoder[Validation.NodeValidationError] =
+      JsonEncoder[String].contramap(e => s"[validation] ${e.msg}")
+    // Encodes an array
+    implicit val onValidationEncoder:        JsonEncoder[OnValidation]                   =
+      JsonEncoder[Chunk[Validation.NodeValidationError]].contramap(v => Chunk.fromIterable(v.validations.toIterable))
+    // Encode strings
+    implicit val onSaveInventoryEncoder:     JsonEncoder[OnSaveInventory]                = JsonEncoder[String].contramap(_.errorMsg)
+    implicit val onSaveNodeEncoder:          JsonEncoder[OnSaveNode]                     = JsonEncoder[String].contramap(_.errorMsg)
+    implicit val onAcceptationEncoder:       JsonEncoder[OnAcceptation]                  = JsonEncoder[String].contramap(_.errorMsg)
+
+    // we need to redefine this because we don't want a discriminator, and also the JSON subtype are not the same
+    implicit val encoder: JsonEncoder[CreationError] = new JsonEncoder[CreationError] {
+      override def unsafeEncode(a: CreationError, indent: Option[Int], out: Write): Unit = a match {
+        case v: OnValidation    => JsonEncoder[OnValidation].unsafeEncode(v, indent, out)
+        case v: OnSaveInventory => JsonEncoder[OnSaveInventory].unsafeEncode(v, indent, out)
+        case v: OnSaveNode      => JsonEncoder[OnSaveNode].unsafeEncode(v, indent, out)
+        case v: OnAcceptation   => JsonEncoder[OnAcceptation].unsafeEncode(v, indent, out)
```

Comment:
Why not  `case v: OnAcceptation   => onAcceptationEncoder.unsafeEncode(v, indent, out)` ? 
Since they are just above ?

Reply:
I'm just used to summon the implicit like this whenever possible :grin: 
Because in scala 3 some day it may become nameless : `given JsonEncoder[OnAcceptation]` (even now in scala 2, it can be derived using `@jsonDerive` macro if we want)